src/Uno.UI/UI/Xaml/Controls/MenuFlyout/CascadingMenuHelper.cs

```diff
@@ -0,0 +1,964 @@
+
+				TimeSpan delayTimeSpan = TimeSpan.FromMilliseconds(m_subMenuShowDelay);
+				m_delayCloseMenuTimer.Interval = delayTimeSpan;
+			}
+		}
+
+		// Handler for the Tick event on the delay close menu timer.
+		void DelayCloseMenuTimerTickHandler()
+		{
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: DelayCloseMenuTimerTickHandler.", this));
+#endif // CMH_DEBUG
+
+			CloseSubMenu();
+
+			if (m_delayCloseMenuTimer != null)
+			{
+#if CMH_DEBUG
+				(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: DelayCloseMenuTimerTickHandler - Stopping m_delayCloseMenuTimer.", this));
+#endif // CMH_DEBUG
+				m_delayCloseMenuTimer.Stop();
+			}
+		}
+
+		// Ensure that any currently open sub menus are closed
+		void EnsureCloseExistingSubItems()
+		{
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: EnsureCloseExistingSubItems.", this));
+#endif // CMH_DEBUG
+
+			ISubMenuOwner ownerAsSubMenuOwner = m_wpOwner?.Target as ISubMenuOwner;
+
+			if (ownerAsSubMenuOwner != null)
+			{
+				ownerAsSubMenuOwner.ClosePeerSubMenus();
+			}
+		}
+
+		internal void SetSubMenuPresenter(FrameworkElement subMenuPresenter)
+		{
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: SetSubMenuPresenter.", this));
+#endif // CMH_DEBUG
+
+			m_wpSubMenuPresenter =new WeakReference(subMenuPresenter);
+
+			ISubMenuOwner ownerAsSubMenuOwner = m_wpOwner?.Target as ISubMenuOwner;
+
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: SetSubMenuPresenter - ownerAsSubMenuOwner=0x%p.", this, ownerAsSubMenuOwner));
+#endif // CMH_DEBUG
+
+			if (ownerAsSubMenuOwner != null)
+			{
+				IMenuPresenter menuPresenter = subMenuPresenter as IMenuPresenter;
+
+#if CMH_DEBUG
+        (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: SetSubMenuPresenter - menuPresenter=0x%p.", this, menuPresenter));
+#endif // CMH_DEBUG
+
+				if (menuPresenter != null)
+				{
+					menuPresenter.Owner = ownerAsSubMenuOwner;
+				}
+			}
+		}
+
+		// Shows the sub menu at the appropriate position.
+		// The sub menu will be adjusted if the sub presenter size changes.
+		internal void OpenSubMenu()
+		{
+			ISubMenuOwner ownerAsSubMenuOwner = m_wpOwner?.Target as ISubMenuOwner;
+
+#if CMH_DEBUG
+    (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OpenSubMenu - ownerAsSubMenuOwner=0x%p.", this, ownerAsSubMenuOwner));
+#endif // CMH_DEBUG
+
+			if (ownerAsSubMenuOwner != null)
+			{
+				ownerAsSubMenuOwner.PrepareSubMenu();
+
+				bool isSubMenuOpen = false;
+				isSubMenuOpen = ownerAsSubMenuOwner.IsSubMenuOpen;
+
+				if (!isSubMenuOpen)
+				{
+					Control ownerAsControl  = m_wpOwner?.Target as Control;
+
+					if (ownerAsControl != null)
+					{
+						EnsureCloseExistingSubItems();
+
+						double subItemWidth = 0;
+						subItemWidth = ownerAsControl.ActualWidth;
+
+						FlowDirection flowDirection = FlowDirection.LeftToRight;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'flowDirection'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)